torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -91,12 +95,18 @@ namespace c10d {
 
 std::string getNcclVersion();
 std::string ncclGetErrorWithVersion(ncclResult_t error);
+// Need this wrapper because ProcessGroupNCCL calls needs to call into C10D_NCCL_CHECK
+// with failureReason as an optional argument.
+void c10d_nccl_check_wrapper(ncclResult_t cmd, c10::optional<std::string> failureReason = c10::nullopt);
```

Comment:
nit: Would it be cleaner to just pass null in all other `C10D_NCCL_CHECK` calls instead of having a wrapper?